metadata-service/restli-servlet-impl/src/main/java/com/linkedin/metadata/resources/entity/EntityV2Resource.java

```diff
@@ -0,0 +1,98 @@
+package com.linkedin.metadata.resources.entity;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.collect.ImmutableSet;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.entity.EntityResponse;
+import com.linkedin.metadata.entity.EntityService;
+import com.linkedin.metadata.restli.RestliUtil;
+import com.linkedin.parseq.Task;
+import com.linkedin.restli.server.annotations.Optional;
+import com.linkedin.restli.server.annotations.QueryParam;
+import com.linkedin.restli.server.annotations.RestLiCollection;
+import com.linkedin.restli.server.annotations.RestMethod;
+import com.linkedin.restli.server.resources.CollectionResourceTaskTemplate;
+import io.opentelemetry.extension.annotations.WithSpan;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+import lombok.extern.slf4j.Slf4j;
+
+import static com.linkedin.metadata.restli.RestliConstants.PARAM_ASPECTS;
+import static com.linkedin.metadata.utils.PegasusUtils.urnToEntityName;
+
+
+/**
+ * Single unified resource for fetching, updating, searching, & browsing DataHub entities
+ */
+@Slf4j
+@RestLiCollection(name = "entitiesV2", namespace = "com.linkedin.entity")
+public class EntityV2Resource extends CollectionResourceTaskTemplate<String, EntityResponse> {
+
+  @Inject
+  @Named("entityService")
+  private EntityService _entityService;
+
+  /**
+   * Retrieves the value for an entity that is made up of latest versions of specified aspects.
+   */
+  @RestMethod.Get
+  @Nonnull
+  @WithSpan
+  public Task<EntityResponse> get(@Nonnull String urnStr,
+      @QueryParam(PARAM_ASPECTS) @Optional @Nullable String[] aspectNames) throws URISyntaxException {
+    log.debug("GET V2 {}", urnStr);
+    final Urn urn = Urn.createFromString(urnStr);
+    return RestliUtil.toTask(() -> {
+      final String entityName = urnToEntityName(urn);
+      final Set<String> projectedAspects =
+          aspectNames == null ? getAllAspectNames(entityName) : new HashSet<>(Arrays.asList(aspectNames));
+      Map<Urn, EntityResponse> entities;
+      try {
+        entities = _entityService.getEntitiesV2(entityName, ImmutableSet.of(urn), projectedAspects);
+      } catch (Exception e) {
+        throw new RuntimeException(
+            String.format("Failed to get entity with urn: %s, aspects: %s", urn, projectedAspects), e);
+      }
+      return entities.get(urn);
+    }, MetricRegistry.name(this.getClass(), "get"));
+  }
+
+  @RestMethod.BatchGet
+  @Nonnull
+  @WithSpan
+  public Task<Map<Urn, EntityResponse>> batchGet(@Nonnull Set<String> urnStrs,
```

Comment:
Just a note for us to discuss later: The downside to using restli batch get here is that there are specific requirements for the batch get method - like providing a list in the query parameters in a super Restli-specific way: GET /entitiesV2?ids=List(id1,id2) or something similar. If we started to use Spring directly we could actually design the API however we want. Definitely something worth considering